diagnostics/app/model/diagnostics/css/Css.scala

```diff
@@ -0,0 +1,33 @@
+package model.diagnostics.css
+
+import play.api.libs.json._
+import collection.Map
+
+case class CssReport(
+  href: String,
+  breakpoint: String,
+  className: String,
+  contentType: String,
+  selectors: JsObject ) {
+
+  override def toString(): String = {
+    val selectorMap = selectors.value.asInstanceOf[Map[String, JsBoolean]]
```

Comment:
It's more a difference of typesafety. You've no proof at this point that
JsObject only contains JsBooleans as values, so if you did this and it
didn't, you'd get a runtime error. `collect` will just discard values that
aren't JsBooleans, so it's safe.

On 15 January 2015 at 11:45, Richard Nguyen <notifications@github.com>
wrote:

> In diagnostics/app/model/diagnostics/css/Css.scala
> <https://github.com/guardian/frontend/pull/7793#discussion-diff-23004646>:
>
> > @@ -0,0 +1,33 @@
> > +package model.diagnostics.css
> > +
> > +import play.api.libs.json._
> > +import collection.Map
> > +
> > +case class CssReport(
> > +  href: String,
> > +  breakpoint: String,
> > +  className: String,
> > +  contentType: String,
> > +  selectors: JsObject ) {
> > +
> > +  override def toString(): String = {
> > +    val selectorMap = selectors.value.asInstanceOf[Map[String, JsBoolean]]
>
> yes! although I'm sure pattern match/ partial functions use asInstanceOf
> too, however they use it correctly!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/7793/files#r23004646>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396